### PR TITLE
Achivement system

### PR DESCRIPTION
**Pull request by Rushabh378**

### CODE REVIEW

File: Assets/Scripts/Ui/Achivment.cs [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        private Animator popUp;
+        private TextMeshProUGUI[] achievementText;
+        private Rank rank;
+        private int enemyCount = 0;
+        private int shootCount = 0;
+
+        private void Start()
+        {
+            popUp = GetComponent<Animator>();
+            achievementText = GetComponentsInChildren<TextMeshProUGUI>();
```
Comment (SahayDarshan) [inline Assets/Scripts/Ui/Achivment.cs:19]: Do not access the child component like this, drag and drop it. There can be multiple child objects containing same objects which can create problems in getting the correct reference.

File: Assets/Scripts/Ui/Achivment.cs [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+            MVC.Enemy.TankController.OnEnemyDeath += EnemyKillAchievement;
+            MVC.Player.TankController.OnPlayerShoot += ShootingAchievement;
+        }
+
```
Comment (SahayDarshan) [inline Assets/Scripts/Ui/Achivment.cs:23]: For now this achievement system will work but what will happen if there are 100 types of achievements. Would you create 100 different methods for all the achievements? No, right. 

So Create a generic method which will handle all types of achievements based on the parameters you pass into it.
Take the help of your clanmates or ask the questions in discord.

**Review by SahayDarshan — COMMENTED** [review]
I have added the review in your achievement script. Please refactor accordingly.

**Comment by Rushabh378** [comment]
thanks for pointing it out I will work on it.

**Pull request merged**